src/main/java/org/tikv/raw/RawKVClient.java

```diff
@@ -79,42 +88,32 @@ public void put(ByteString key, ByteString value) {
     }
   }
 
+  public void batchPut(List<ByteString> keys, List<ByteString> values) {
+    batchPut(ConcreteBackOffer.newRawKVBackOff(), keys, values);
+  }
+
   /**
    * Put a list of raw key-value pair to TiKV
    *
-   * @param kvPairs kvPairs
+   * @param keys keys
+   * @param values values
    */
-  public void batchPut(List<Kvrpcpb.KvPair> kvPairs) {
-    BackOffer backOffer = defaultBackOff();
-    List<Kvrpcpb.KvPair> remainingPairs = new ArrayList<>();
-    while (true) {
-      kvPairs.addAll(remainingPairs);
-      remainingPairs.clear();
-      Map<Pair<TiRegion, Metapb.Store>, List<Kvrpcpb.KvPair>> regionMap = new HashMap<>();
-      for (Kvrpcpb.KvPair kvPair : kvPairs) {
-        Pair<TiRegion, Metapb.Store> pair = regionManager.getRegionStorePairByKey(kvPair.getKey());
-        regionMap.computeIfAbsent(pair, t -> new ArrayList<>()).add(kvPair);
-      }
+  public void batchPut(BackOffer backOffer, List<ByteString> keys, List<ByteString> values) {
+    assert keys.size() == values.size();
```

Comment:
Use guava's assert. Unless manually turned on, assert keyword will not be active in production. What you actually need here is indeed an exception.
Or consider use something like Pair then you don't need such check / or a Map since a key should not be insert twice.

Reply:
done.